CMakeLists.txt

```diff
@@ -46,33 +40,37 @@ endif ()
 
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
-if (NOT ("${GOOGLE_CLOUD_CPP_CXX_STANDARD}" STREQUAL ""))
-    message(
-        WARNING
-            "GOOGLE_CLOUD_CPP_CXX_STANDARD is retired, use CMAKE_CXX_STANDARD")
-endif ()
-
 if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
 
-    # GCC >= 5.4 is required by Abseil, so we must require it too.
-    if (CMAKE_CXX_COMPILER_VERSION VERSION_LESS 5.4)
+    # GCC >= 5.1 is required by Abseil, so we must require it too. We only test
+    # with >= 6.3 because that is easy to install in our CI systems.
+    if (CMAKE_CXX_COMPILER_VERSION VERSION_LESS 5.1)
```

Comment:
AFAIK, it is not a C++11 thing.  While no compiler is perfect, GCC 5.0 has pretty decent support for C++11.  Hmm... maybe you know something I don't, is it that Abseil ran into C++11 bugs with GCC 5.0?  I thought they had simply decided to support 5.1+ because time has passed:

https://abseil.io/blog/20201001-platforms
https://abseil.io/docs/cpp/platforms/platforms

Reply:
The comment referenced Abseil needing >= 5.1. The fatal message referenced older versions lacking C++11 support, but I missed that the message also contained "or are not supported by some of our dependencies".

I don't think we should have any checks here that are only because of our dependencies. That's up to our deps to enforce. Also, I don't think we should try to keep track of all these details about our dependencies. For example, what if Abseil changed their mind and started working with GCC 5.0? We'd still fatally fail builds that should otherwise work.

My vote for these checks would be to only check the versions that _google-cloud-cpp_ claims support for or tests with. And we can warn if using a compiler earlier than we test with. That seems like it covers all cases to me, without us having to know specific details that might change about our deps.

But to be clear, it's also fine with me if you decide to leave these checks.

Reply:
Ack.  Here is an interesting case for us to think about: protobuf compiles with Clang 3.8, but the code protoc produces from the googleapis protos does not compile with Clang 3.8.  Should we let the configuration proceed when we know the build will fail?

Reply:
FWIW, I appreciate your comments here, they are helping me think through this.

Reply:
In that case, we only know the build will fail today. We don't know that protobuf won't fix the generated code to work with clang 3.8 tomorrow. And so I'd still say that I'm certainly happy warning about any version that's older than we test with. But beyond that, our fatal errors are informed guesses about (undocumented?) details about our dependencies. My vote would be to only issue warnings/checks based on _our own_ guarantees.

The default action here is to keep them because it takes effort to submit a new PR to remove them. I don't want to put in that effort. And if you don't  either, sticking w/ the current fatal warnings is fine with me.